manim/scene/scene.py

```diff
@@ -828,6 +840,65 @@ def update_skipping_status(self):
                 file_writer_config['skip_animations'] = True
                 raise EndSceneEarlyException()
 
+    def handle_caching_play(func): 
+        """
+        This method is used internally to wrap the passed function into a function that will compute the hash of the play invocation, 
+        and will act accordingly : either skip the animation because already cached, either nothing and let the play invocation be processed normally.
+
+        Parameters
+        ----------
+        *args : 
+            Animation or mobject with mobject method and params
+
+        **kwargs : 
+            named parameters affecting what was passed in *args e.g run_time, lag_ratio etc.
+        """
+        def wrapper(self, *args, **kwargs):
+            self.revert_to_original_skipping_status()
+            animations = self.compile_play_args_to_animation_list(
+                *args, **kwargs
+                )
+            self.add_mobjects_from_animations(animations)
+            if not file_writer_config['disable_caching']:
+                mobjects_on_scene = self.get_mobjects()
+                hash_play = get_hash_from_play_call(self.__dict__['camera'], animations, mobjects_on_scene)
```

Comment:
Any particular reason for using `self.__dict__['camera']` instead of `self.camera`?

```suggestion
                hash_play = get_hash_from_play_call(self.camera, animations, mobjects_on_scene)
```